src/ZoneServer/Skills/Handlers/Swordsmen/Barbarian/Barbarian_StompingKick.cs

```diff
@@ -0,0 +1,121 @@
+		/// Handles skill, damaging targets.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		public async void Handle(Skill skill, ICombatEntity caster, Position originPos, Position farPos, ICombatEntity target)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.TurnTowards(farPos);
+			caster.SetAttackState(true);
+
+			var splashWidth = 55;
+
+			// Wild Nature effects - +2 range per stack
+			if (caster.TryGetBuff(BuffId.ScudInstinct_Buff, out var wildNature))
+			{
+				splashWidth += 2 * wildNature.OverbuffCounter;
+			}
+
+			var splashParam = skill.GetSplashParameters(caster, originPos, farPos, length: 0, width: splashWidth, angle: 0);
+			var splashArea = skill.GetSplashArea(SplashType.Circle, splashParam);
+
+			Send.ZC_SKILL_READY(caster, skill, originPos, farPos);
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, farPos, null);
+
+			// wait for the user to hit the ground
+
+			if (caster.Components.TryGet<MovementComponent>(out var movementComponent))
+			{
+				while (!movementComponent.IsGrounded)
+				{
+					await Task.Delay(TimeSpan.FromMilliseconds(30));
+				}
+			}
+			CallSafe(this.Attack(skill, caster, splashArea));
+		}
+
+		/// <summary>
+		/// Executes the actual attack after a delay.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async Task Attack(Skill skill, ICombatEntity caster, ISplashArea splashArea)
+		{
+			var hitDelay = TimeSpan.FromMilliseconds(100);
+			var damageDelay = TimeSpan.FromMilliseconds(50);
+			var skillHitDelay = TimeSpan.Zero;
+
+			//await Task.Delay(hitDelay);
+
+			var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+			var hits = new List<SkillHitInfo>();
+
+			foreach (var target in targets.LimitBySDR(caster, skill))
+			{
+				var modifier = SkillModifier.Default;
+
+				// Wild Nature effects - 6% damage per stack
+				if (caster.TryGetBuff(BuffId.ScudInstinct_Buff, out var wildNature))
+				{
+					modifier.DamageMultiplier += 0.06f * wildNature.OverbuffCounter;
+				}
+
+				var skillHitResult = SCR_SkillHit(caster, target, skill, modifier);
+				target.TakeDamage(skillHitResult.Damage, caster);
+
+				var skillHit = new SkillHitInfo(caster, target, skill, skillHitResult, damageDelay, skillHitDelay);
+				skillHit.HitEffect = HitEffect.Impact;
+
+				hits.Add(skillHit);
+			}
+
+			Send.ZC_SKILL_HIT_INFO(caster, hits);
+
+			// Barbarian 31 triggers a rejump afterwards
+			if (caster.IsAbilityActive(AbilityId.Barbarian31))
+			{
+				await Task.Delay(TimeSpan.FromMilliseconds(100));
+				if (caster.Components.TryGet<MovementComponent>(out var movementComponent))
+				{
+					// TODO: This doesn't work, may need another way to trigger a jump
```

Comment:
Can you tell me more about what you need to do here?

Reply:
It's supposed to trigger you to jump again, like in a platformer when you bounce off an enemy.  I haven't found any packet I can send for it, though (this ability is also removed).